projects/packages/image-cdn/src/class-image-cdn.php

```diff
@@ -281,9 +281,40 @@ public static function filter_photon_norezise_maybe_inject_sizes_api( $sizes, $a
 	 * * IN-CONTENT IMAGE MANIPULATION FUNCTIONS
 	 **/
 
+	/**
+	 * Given a `class` attribute value, return the first class name with
+	 * the given prefix, excluding that prefix in the return value.
+	 *
+	 * @since $$next-version$$
+	 *
+	 * Example:
+	 *
+	 *     'full-width' = self::parse_class_name_prefixed_by( 'id-5 size-full-width duotone-gray size-wide', 'size-' );
+	 *     null         = self::parse_class_name_prefixed_by( 'id-5 size-full-width duotone-gray size-wide', 'lazy-' );
+	 *     'id-5'       = self::parse_class_name_prefixed_by( 'id-5 size-full-width duotone-gray size-wide', '' );
+	 *
+	 * @param string $class  Value of an HTML `class` attribute.
+	 * @param string $prefix A returned class name must start with this prefix.
+	 * @return string|null part of a class name that follows the given prefix if found, else null.
+	 */
+	public static function parse_class_name_prefixed_by( $class, $prefix ) {
+		if ( ! is_string( $class ) ) {
+			return null;
+		}
+
+		$search_pattern = sprintf( '#(?:^|[ \t\f\r\n])%s(?P<value>[^ \t\f\r\n]+)#', preg_quote( $prefix, null ) );
```

Comment:
good question! I'm so used to the HTML API where we've been very explicit about these that I didn't think about it, but [the PHP docs](https://www.php.net/manual/en/regexp.reference.escape.php) suggest that `\s` may be affected by locale options. I think it would be just as good to leave this as-is, though `\s` is unlikely to be a big problem in practice so if you prefer it I'm happy to make that change.

apparently it may include additional unicode characters representing whitespace if those locale rules are active, which would diverge from how the CSS class names are parsed in the browser.

Reply:
Unicode can't let anything be simple, not even whitespace.
What you've written would be closer to the HTML spec, so let's stick with how you've done it.